### DISCUSSION TITLE
How to get viewer count of a path?

### DISCUSSION THREAD

**Discussion by FurretUber** [bluenviron/mediamtx#2881] [Questions]
### Question

Hi, is there a way to get the viewer count of a path? From what I'm looking at the API, using the `/v3/paths/list`: the `items` have a `source` and `readers` elements, which show the RTMP, RTSP and WebRTC in what seem to be a consistent way: every connection watching the stream with one of these three protocols is shown as an element in `readers`. If I got this right, the number of readers of a path is the number of viewers of said path?

However, this doesn't seem to work with HLS. With HLS, it shows a `hlsMuxer`, which doesn't seem to change, no matter the amount of viewers. I tried to load test with Apache JMeter 60 simultaneous connections via HLS and the `hlsMuxer` did not change (which is not really surprising as it's a count of muxers).

Is checking the amount of `readers` of a path the correct way to get the amount of viewers of a path? And is there a way to get the viewer count of a HLS stream via API?

**Comment by aler9** [comment]
Answer is here: https://github.com/bluenviron/mediamtx/issues/962#issuecomment-1143654570

[a comment by an automated account is not shown]
